### ISSUE TITLE
TypeError: Cannot read property 'type' of undefined  at dbTypeMatchesObjectType

### ISSUE THREAD

**Issue by prashanthmadi** [ParsePlatform/parse-server#3477]
I was getting below error when i try to add a new column to my class in dashboard with parse-server@2.3.2

```
2017-02-04T01:57:00.321Z - Uncaught internal server error. TypeError: Cannot read property 'type' of undefined
    at dbTypeMatchesObjectType (D:\home\site\wwwroot\node_modules\parse-server\lib\Controllers\SchemaController.js:309:13)
    at D:\home\site\wwwroot\node_modules\parse-server\lib\Controllers\SchemaController.js:691:16
    at process._tickDomainCallback (internal/process/next_tick.js:129:7) TypeError: Cannot read property 'type' of undefined
    at dbTypeMatchesObjectType (D:\home\site\wwwroot\node_modules\parse-server\lib\Controllers\SchemaController.js:309:13)
    at D:\home\site\wwwroot\node_modules\parse-server\lib\Controllers\SchemaController.js:691:16
    at process._tickDomainCallback (internal/process/next_tick.js:129:7)
```

**Comment by flovilmart** [comment]
Can you provide the server logs when running with VERBOSE=1?

**Comment by adambarghouthi** [comment]
I'm experiencing this too 

error: Uncaught internal server error. [TypeError: Cannot read property 'type' of undefined] TypeError: Cannot read property 'type' of undefined
    at dbTypeMatchesObjectType (D:\home\site\wwwroot\node_modules\parse-server\lib\Controllers\SchemaController.js:307:13)
    at D:\home\site\wwwroot\node_modules\parse-server\lib\Controllers\SchemaController.js:687:16
    at process._tickDomainCallback (node.js:412:9)

TypeError: Cannot read property 'type' of undefined
    at dbTypeMatchesObjectType (D:\home\site\wwwroot\node_modules\parse-server\lib\Controllers\SchemaController.js:307:13)
    at D:\home\site\wwwroot\node_modules\parse-server\lib\Controllers\SchemaController.js:687:16
    at process._tickDomainCallback (node.js:412:9)

**Comment by adambarghouthi** [comment]
@prashanthmadi are you on Azure also or no?

**Comment by prashanthmadi** [comment]
@adambarghouthi  Yes. I will work on this issue today and update.

**Comment by adambarghouthi** [comment]
@prashanthmadi that'd be great. I'm also in contact with Azure since this issue started happening only after they fixed another issue I was having. So it could be on their end. We'll update this thread if anything

**Comment by prashanthmadi** [comment]
@adambarghouthi issue looks to be related with documentdb. Try with mlab mongodb and it would work...
I'm working with documentdb team and would update you soon.

FYI, I'm part of Azure Support.

**Comment by adambarghouthi** [comment]
@prashanthmadi thank you! Will be waiting for the update

**Comment by prashanthmadi** [comment]
issue is identified and the Azure DocumentDB team will be deploying a new build starting tomorrow.

**Comment by adambarghouthi** [comment]
@prashanthmadi awesome thanks man!

**Comment by swar-it** [comment]
I am still having this issue. Does anyone if it has already been resolved by the Azure DocumentDB team or not?

**Comment by adambarghouthi** [comment]
@swar-it They were deploying the fix yesterday but I'm also still facing this issue

**Comment by prashanthmadi** [comment]
@swar-it @adambarghouthi  I will check with Azure documentdb team and update

**Comment by kerseyr** [comment]
Hi @prashanthmadi did you get any update back?

**Comment by prashanthmadi** [comment]
DocumentDB Team has started deploying fix and it might go till Tuesday. 

A workaround would be to update the _SCHEMA document manually and add a new property using [DocumentDB Studio](https://github.com/mingaliu/DocumentDBStudio/releases) (which directly talks to our backend).

If you have any queries, Feel Free to reach out Siddhesh.Vethe@microsoft.com
